stdlib/src/collections/counter.mojo

```diff
@@ -76,3 +82,400 @@ struct Counter[V: KeyElement](
             count: The count to store in the Counter.
         """
         self._data[value] = count
+
+    fn __iter__(ref [_]self: Self) -> _DictKeyIter[V, Int, __lifetime_of(self)]:
+        """Iterate over the keyword dict's keys as immutable references.
+
+        Returns:
+            An iterator of immutable references to the Counter values.
+        """
+        return self._data.__iter__()
+
+    # ===-------------------------------------------------------------------===#
+    # Trait implementations
+    # ===-------------------------------------------------------------------===#
+
+    fn __len__(self) -> Int:
+        """The number of elements currently stored in the Counter."""
+        return len(self._data)
+
+    fn __bool__(self) -> Bool:
+        """Check if the Counter is empty or not.
+
+        Returns:
+            `False` if the Counter is empty, `True` if there is at least one element.
+        """
+        return bool(len(self))
+
+    # ===-------------------------------------------------------------------===#
+    # Comparison operators
+    # ===-------------------------------------------------------------------===#
+
+    fn __eq__(self, other: Self) -> Bool:
+        """Check if all counts agree. Missing counts are treated as zero.
+
+        Args:
+            other: The other Counter to compare to.
+
+        Returns:
+            True if the two Counters are equal, False otherwise.
+        """
+
+        @parameter
+        @always_inline
+        fn is_eq(keys: _DictKeyIter[V, Int, _]) -> Bool:
+            for e_ref in keys:
+                var e = e_ref[]
+                if self.get(e, 0) != other.get(e, 0):
+                    return False
+            return True
+
+        return is_eq(self.keys()) and is_eq(other.keys())
```

Comment:
`is_eq` captures `other`, so `is_eq(other.keys())` doesn't do anything, right?

Reply:
This is the Mojo version of the Python `Counter.__eq__()` that is based on two list comprehension joined, so a double loop:

![image](https://github.com/modularml/mojo/assets/136875/3b22d908-af0e-46fd-a7da-ab97c53d347d)

It's needed to loop for both the `self` and `other` keys, as the keys listing could not be overlapped.

Reply:
Ah, I misread the code.